tensordict/nn/common.py

```diff
@@ -54,6 +56,17 @@ def _check_all_nested(list_of_keys):
         _nested_key_type_check(key)
 
 
+def dispatch_kwargs(func):
+    @functools.wraps(func)
+    def wrapper(self, tensordict=None, *args, **kwargs):
+        if tensordict is None:
+            tensordict = make_tensordict(**kwargs)
```

Comment:
We should check that all the in_keys are in the kwargs 
Any extra kwarg should be passed to the module forward

eg in the test 
`forward(a=tensor, c=True)`
Would pass the "c" kwarg to the forward of the module and build a TD with "a".

Reply:
Should we throw an exception of not all in_keys are in the kwargs? TensorDictModule.forward() allow now such a case:

```
    def forward(
        self,
        tensordict: TensorDictBase,
        tensordict_out: Optional[TensorDictBase] = None,
        **kwargs,
    ) -> TensorDictBase:
...
        tensors = tuple(tensordict.get(in_key, None) for in_key in self.in_keys)
```

It uses None as a default value if tensordict does not have a key from `self.in_keys`.